tests/test_train_cli.py

```diff
@@ -0,0 +1,155 @@
+"""Test train commandline interface."""
+
+from pathlib import Path
+import shutil
+
+import pytest
+from typer.testing import CliRunner
+
+try:
+    from mace.cli.run_train import run as run_train  # pylint: disable=unused-import
+
+    MACE_IMPORT_ERROR = False
+except ImportError:
+    MACE_IMPORT_ERROR = True
+
+from janus_core.cli.janus import app
+
+DATA_PATH = Path(__file__).parent / "data"
+
+runner = CliRunner()
+
+
+@pytest.mark.skipif(MACE_IMPORT_ERROR, reason="Requires updated version of MACE")
+def test_help():
+    """Test calling `janus train --help`."""
+    result = runner.invoke(app, ["train", "--help"])
+    assert result.exit_code == 0
+    assert "Usage: janus train [OPTIONS]" in result.stdout
+
+
+@pytest.mark.skipif(MACE_IMPORT_ERROR, reason="Requires updated version of MACE")
+def test_train():
+    """Test MLIP training."""
+    model = "test.model"
+    compiled_model = "test_compiled.model"
+    logs_path = "logs"
+    results_path = "results"
+    checkpoints_path = "checkpoints"
+
+    assert not Path(model).exists()
+    assert not Path(compiled_model).exists()
+    assert not Path(logs_path).exists()
+    assert not Path(results_path).exists()
+    assert not Path(checkpoints_path).exists()
+
+    config = DATA_PATH / "mlip_train.yml"
+
+    result = runner.invoke(
+        app,
+        [
+            "train",
+            "--mlip-config",
+            config,
+        ],
+    )
+    try:
+        assert Path(model).exists()
+        assert Path(compiled_model).exists()
+        assert Path(logs_path).is_dir()
+        assert Path(results_path).is_dir()
+        assert Path(checkpoints_path).is_dir()
+    finally:
+        # Tidy up models
+        Path(model).unlink(missing_ok=True)
+        Path(compiled_model).unlink(missing_ok=True)
+
+        # Tidy up directories
+        shutil.rmtree(logs_path, ignore_errors=True)
+        shutil.rmtree(results_path, ignore_errors=True)
+        shutil.rmtree(checkpoints_path, ignore_errors=True)
+
+        assert result.exit_code == 0
+
+
+@pytest.mark.skipif(MACE_IMPORT_ERROR, reason="Requires updated version of MACE")
+def test_train_with_foundation():
+    """Test MLIP training raises error with foundation_model in config."""
+    config = DATA_PATH / "mlip_train_invalid.yml"
+
+    result = runner.invoke(
+        app,
+        [
+            "train",
+            "--mlip-config",
+            config,
+        ],
+    )
+    assert result.exit_code == 1
+    assert isinstance(result.exception, ValueError)
+
+
+@pytest.mark.skipif(MACE_IMPORT_ERROR, reason="Requires updated version of MACE")
+def test_fine_tune():
+    """Test MLIP fine-tuning."""
+    model = "test-finetuned.model"
+    compiled_model = "test-finetuned_compiled.model"
+    logs_path = "logs"
+    results_path = "results"
+    checkpoints_path = "checkpoints"
```

Comment:
Been playing around in other repos with [`tempfile.TemporaryDirectory`](https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryDirectory) which may be what we want with these rather than the `try`-`finally`s. Probably better to do globally as another PR though.

Reply:
In this case, MACE is creating these directories, so even if they're options that we can set, while we're reliant on passing a predefined config file, this wouldn't be particularly straightforward to make use.

Worth looking into though, I agree.